### ISSUE TITLE
$query->order(<string>) SQL broken when quoting identifiers

### ISSUE THREAD

**Issue by Schlaefer** [cakephp/cakephp#6805]
On dev-master 563db7256f5e52aa3db04b9e79d23c4bb38cb0d9. I try to give a test example: https://github.com/Schlaefer/cakephp/commit/ae26da6cca4ab4fdd74a689f6df0f966f130f49b

```
// query
$query->order('something ASC')

// should result in (?)
ORDER BY "something" ASC

// is
ORDER BY "something" AS "C"
```

3.0.6 didn't quote anything, but the unquoted SQL was O.K.

**Comment by markstory** [comment]
I'll fix this as it probably broke when I fixed `order(['Model.name' => 'ASC'])` with identifier quoting enabled.

**Comment by markstory** [comment]
Fixed with #6806.
